client/routing.go

```diff
@@ -48,7 +48,14 @@ func (c *Client) List(ctx context.Context, req *routingv1.ListRequest) (<-chan *
 			}
 
 			// Stream ListResponse directly (no legacy wrapper)
-			resCh <- obj
+			// Use select to prevent goroutine leak if consumer stops reading
+			select {
+			case resCh <- obj:
+			case <-ctx.Done():
+				logger.Error("context cancelled while receiving list response", "error", ctx.Err())
+
+				return
+			}
```

Comment:
We create our stream with the passed context in the method. Once the context closes, so does the stream. Calling `stream.Recv` on a stream with expired context will return an error and we will exit. 

This means that the context check here is not needed as its already handled by the stream itself. Same for other methods.